R/pkg/R/types.R

```diff
@@ -67,3 +67,27 @@ rToSQLTypes <- as.environment(list(
   "double" = "double",
   "character" = "string",
   "logical" = "boolean"))
+
+# Helper function of coverting decimal type. When backend returns column type in the
+# format of decimal(,) (e.g., decimal(10, 0)), this function coverts the column type
+# as double type. This function converts backend returned types that are not the key
+# of PRIMITIVE_TYPES, but should be treated as PRIMITIVE_TYPES.
+# @param A type returned from the JVM backend.
+# @return A type is the key of the PRIMITIVE_TYPES.
+specialtypeshandle <- function(type) {
+  if (!is.null(PRIMITIVE_TYPES[[type]])) {
+    type
+  } else {
+    firstChar <- substr(type, 1, 1)
+    returntype <- NULL
+    switch (firstChar,
+    d = {
+      m <- regexec("^decimal(.+)", type)
```

Comment:
I'd just simplify this at this point. It would be easy to add the additional checks later